tests/unit/extensions/test_qt_framework.py

```diff
@@ -0,0 +1,249 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2023 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import pytest
+
+from snapcraft.extensions import qt_framework
+from snapcraft.extensions.extension import get_extensions_data_dir
+
+_EXTENSION_NAME = "qt6-5"
```

Comment:
Do you mind testing `qt5-15` and `qt6-5`?  You should be able to parameterize the fixtures like this:
https://docs.pytest.org/en/7.3.x/how-to/fixtures.html#parametrizing-fixtures